brain/python/circuit.cpp

```diff
@@ -0,0 +1,181 @@
+GET_CIRCUIT_PROPERTY_FOR_GIDS(ElectrophysiologyTypes)
+
+bp::object Circuit_getMorphologyURIs( const Circuit& circuit, bp::object gids )
+{
+    return toPythonList( circuit.getMorphologyURIs( gidsFromPython( gids )));
+}
+
+#define GET_CIRCUIT_PROPERTY_VALUES(property) \
+    bp::list Circuit_get##property( const Circuit& circuit )\
+{                                                           \
+    return toPythonList( circuit.get##property( ));         \
+}
+GET_CIRCUIT_PROPERTY_VALUES(MorphologyNames)
+GET_CIRCUIT_PROPERTY_VALUES(ElectrophysiologyNames)
+
+bp::list Circuit_loadMorphologies( const Circuit& circuit,
+                                   bp::object gids, Circuit::Coordinates coords )
+{
+    return toPythonList(
+        circuit.loadMorphologies( gidsFromPython( gids ), coords ));
+}
+
+bp::object Circuit_getPositions( const Circuit& circuit, bp::object gids )
+{
+    return toNumpy( circuit.getPositions( gidsFromPython( gids )));
+}
+
+bp::object Circuit_getTransforms( const Circuit& circuit, bp::object gids )
+{
+    bp::object matrices =
+        toNumpy( circuit.getTransforms( gidsFromPython( gids )));
+    // We want the result to be indexed using regular mathematical notation
+    // even if the actual storage is column-major.
+    return matrices.attr( "transpose" )( 0, 2, 1 );
+}
+
+bp::object Circuit_getRotations( const Circuit& circuit, bp::object gids )
+{
+    return toNumpy( circuit.getRotations( gidsFromPython( gids )));
+}
+
+SynapsesWrapper Circuit_getAfferentSynapses(
+    const CircuitPtr& circuit, bp::object gids,
+    const brain::SynapsePrefetch prefetch )
+{
+    return SynapsesWrapper( circuit->getAfferentSynapses( gidsFromPython( gids ),
+                                                          prefetch ),
+                            circuit );
+}
+
+SynapsesWrapper Circuit_getEfferentSynapses(
+    const CircuitPtr& circuit, bp::object gids,
+    const brain::SynapsePrefetch prefetch )
+{
+    return SynapsesWrapper( circuit->getEfferentSynapses( gidsFromPython( gids ),
+                                                          prefetch ),
+                            circuit );
+}
+
+SynapsesWrapper Circuit_getProjectedSynapses(
+    const CircuitPtr& circuit, bp::object pre, bp::object post,
+    const brain::SynapsePrefetch prefetch )
+{
+    return SynapsesWrapper( circuit->getProjectedSynapses(
+                                gidsFromPython( pre ), gidsFromPython( post ),
+                                prefetch ),
+                            circuit );
+}
+
+}
+
+void export_Circuit()
+{
+
+bp::class_< Circuit, boost::noncopyable, CircuitPtr >
+    circuitWrapper( "Circuit", bp::no_init );
+
+bp::scope circuitScope = circuitWrapper;
+
+bp::enum_< Circuit::Coordinates >( "Coordinates" )
+    .value( "global_", Circuit::Coordinates::global )
+    .value( "local", Circuit::Coordinates::local )
+    ;
+
+circuitWrapper
+    .def( "__init__", bp::make_constructor( Circuit_initFromURI ))
```

Comment:
these all look fine to me, and using functions instead of properties is also fine w/ me.

I might call num_neurons 'neuron_count', but that doesn't matter